### PR TITLE
Create enum ElementType directory-server

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

**Review by YenguiSeddik — APPROVED** [review]
test: ok

File: src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public class ElementAttributes {
     private String lastModifiedBy;
 
     public boolean isAllowed(@NonNull String userId) {
-        if (!type.equals(DIRECTORY)) {
+        if (!type.equals(ElementType.DIRECTORY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java:55]: type != ElementType.DIRECTORY

File: src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java [hunk 2]
```diff
@@ -88,7 +89,7 @@ public boolean isAttributesUpdatable(@NonNull ElementAttributes newElementAttrib
             Objects.nonNull(newElementAttributes.getDescription()) ||
             StringUtils.isNotBlank(newElementAttributes.getElementName()) ||
                     //Only the owner can update the accessRights of a directory (to avoid user locking themselves out of a directory they don't own
-                    type.equals(DIRECTORY) && Objects.nonNull(newElementAttributes.getAccessRights()) && userId.equals(owner))
+                    type.equals(ElementType.DIRECTORY) && Objects.nonNull(newElementAttributes.getAccessRights()) && userId.equals(owner))
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/repository/DirectoryElementEntity.java:92]: type == ElementType.DIRECTORY

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 3]
```diff
@@ -108,7 +103,7 @@ public ElementAttributes createElement(ElementAttributes elementAttributes, UUID
         assertElementNotExist(parentDirectoryUuid, elementAttributes.getElementName(), elementAttributes.getType());
         assertAccessibleDirectory(parentDirectoryUuid, userId);
         DirectoryElementEntity elementEntity = insertElement(elementAttributes, parentDirectoryUuid);
-        var isCurrentElementPrivate = elementAttributes.getType().equals(DIRECTORY) ? elementAttributes.getAccessRights().getIsPrivate() : null;
+        var isCurrentElementPrivate = elementAttributes.getType().equals(ElementType.DIRECTORY) ? elementAttributes.getAccessRights().getIsPrivate() : null;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:106]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 4]
```diff
@@ -150,7 +145,7 @@ private DirectoryElementEntity insertElement(ElementAttributes elementAttributes
                         parentDirectoryUuid,
                         elementAttributes.getElementName(),
                         elementAttributes.getType(),
-                        elementAttributes.getType().equals(DIRECTORY) ? elementAttributes.getAccessRights().getIsPrivate() : null,
+                        elementAttributes.getType().equals(ElementType.DIRECTORY) ? elementAttributes.getAccessRights().getIsPrivate() : null,
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:148]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 5]
```diff
@@ -220,65 +215,65 @@ public void createElementInDirectoryPath(String directoryPath, ElementAttributes
...
 
-    private Stream<ElementAttributes> getAllDirectoryElementsStream(UUID directoryUuid, List<String> types, String userId, boolean stashed) {
+    private Stream<ElementAttributes> getAllDirectoryElementsStream(UUID directoryUuid, List<ElementType> types, String userId, boolean stashed) {
         LocalDateTime stashDate = stashed ? getDirectoryElementEntity(directoryUuid).getStashDate() : null;
         List<DirectoryElementEntity> directoryElements = repositoryService.findAllByParentIdAndStashedAndStashDate(directoryUuid, stashed, stashDate);
         Map<UUID, Long> subdirectoriesCountsMap = getSubDirectoriesCountMap(userId, types, directoryElements);
         return directoryElements
                 .stream()
-                .filter(e -> e.getType().equals(DIRECTORY) || types.isEmpty() || types.contains(e.getType()))
+                .filter(e -> e.getType().equals(ElementType.DIRECTORY) || types.isEmpty() || types.contains(e.getType()))
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:264]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 6]
```diff
@@ -308,7 +303,7 @@ public void updateElement(UUID elementUuid, ElementAttributes newElementAttribut
         );
 
         //true if we updated a study name
-        if (elementEntity.getType().equals(STUDY) && StringUtils.isNotBlank(newElementAttributes.getElementName())) {
+        if (elementEntity.getType().equals(ElementType.STUDY) && StringUtils.isNotBlank(newElementAttributes.getElementName())) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:306]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 7]
```diff
@@ -344,7 +339,7 @@ private void updateElementDirectory(UUID elementUuid, UUID newDirectoryUuid, Str
     }
 
     private void validateElementForUpdate(DirectoryElementEntity element, UUID newDirectoryUuid, String userId) {
-        if (element.getType().equals(DIRECTORY)) {
+        if (element.getType().equals(ElementType.DIRECTORY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:342]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 8]
```diff
@@ -365,32 +360,32 @@ private void emitDirectoryChangedNotifications(DirectoryElementEntity element, D
     }
 
     private void notifyStudyUpdateIfApplicable(DirectoryElementEntity element, String userId) {
-        if (element.getType().equals(STUDY)) {
+        if (element.getType().equals(ElementType.STUDY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:363]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 9]
```diff
@@ -365,32 +360,32 @@ private void emitDirectoryChangedNotifications(DirectoryElementEntity element, D
...
         }
     }
 
     private void validateNewDirectory(UUID newDirectoryUuid) {
         DirectoryElementEntity newDirectory = repositoryService.getElementEntity(newDirectoryUuid)
-                .orElseThrow(() -> DirectoryException.createElementNotFound(DIRECTORY, newDirectoryUuid));
+                .orElseThrow(() -> DirectoryException.createElementNotFound(ElementType.DIRECTORY.name(), newDirectoryUuid));
 
-        if (!newDirectory.getType().equals(DIRECTORY)) {
+        if (!newDirectory.getType().equals(ElementType.DIRECTORY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:372]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 10]
```diff
@@ -365,32 +360,32 @@ private void emitDirectoryChangedNotifications(DirectoryElementEntity element, D
...
+    private boolean directoryHasElementOfNameAndType(UUID directoryUUID, String userId, String elementName, ElementType elementType) {
         return getDirectoryElementsStream(directoryUUID, userId, List.of(elementType))
             .anyMatch(
                 e -> e.getElementName().equals(elementName)
             );
     }
 
     private boolean isElementUpdatable(ElementAttributes element, String userId, boolean forDeletion) {
-        if (element.getType().equals(DIRECTORY)) {
+        if (element.getType().equals(ElementType.DIRECTORY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:385]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 11]
```diff
@@ -739,7 +734,7 @@ public void deleteElements(List<UUID> elementsUuid, String userId) {
                     parentUuid == null,
                     parentUuid == null ? NotificationType.DELETE_DIRECTORY : NotificationType.UPDATE_DIRECTORY
             );
-            if (STUDY.equals(entity.getType())) {
+            if (ElementType.STUDY.equals(entity.getType())) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:737]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 12]
```diff
@@ -678,7 +673,7 @@ public void stashElements(List<UUID> elementsUuid, String userId) {
                     result.addAll(descendants);
                     return result.stream().map(e -> {
                         DirectoryElementEntity stashedElement = e.stashElement(true, stashDate);
-                        if (Objects.equals(e.getType(), STUDY)) {
+                        if (Objects.equals(e.getType(), ElementType.STUDY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:676]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 13]
```diff
@@ -560,7 +555,7 @@ public void emitDirectoryChangedNotification(UUID elementUuid, String userId) {
 
     public void areElementsAccessible(@NonNull String userId, @NonNull List<UUID> elementUuids) {
         getElements(elementUuids, true, List.of()).stream()
-                .map(e -> e.getType().equals(DIRECTORY) ? e : getParentElement(e.getElementUuid()))
+                .map(e -> e.getType().equals(ElementType.DIRECTORY) ? e : getParentElement(e.getElementUuid()))
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:558]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 14]
```diff
@@ -449,7 +444,7 @@ public List<ElementAttributes> getPath(UUID elementUuid, String userId) {
         }
         DirectoryElementEntity currentElement = currentElementOpt.get();
 
-        if (currentElement.getType().equals(DIRECTORY)) {
+        if (currentElement.getType().equals(ElementType.DIRECTORY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:447]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 15]
```diff
@@ -421,11 +416,11 @@ public void deleteElement(UUID elementUuid, String userId) {
     }
 
     private void deleteElement(ElementAttributes elementAttributes, String userId) {
-        if (elementAttributes.getType().equals(DIRECTORY)) {
+        if (elementAttributes.getType().equals(ElementType.DIRECTORY)) {
             deleteSubElements(elementAttributes.getElementUuid(), userId);
         }
         repositoryService.deleteElement(elementAttributes.getElementUuid());
-        if (STUDY.equals(elementAttributes.getType())) {
+        if (ElementType.STUDY.equals(elementAttributes.getType())) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:423]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 16]
```diff
@@ -421,11 +416,11 @@ public void deleteElement(UUID elementUuid, String userId) {
     }
 
     private void deleteElement(ElementAttributes elementAttributes, String userId) {
-        if (elementAttributes.getType().equals(DIRECTORY)) {
+        if (elementAttributes.getType().equals(ElementType.DIRECTORY)) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:419]: Use ==

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 17]
```diff
@@ -365,32 +360,32 @@ private void emitDirectoryChangedNotifications(DirectoryElementEntity element, D
...
             );
     }
 
     private boolean isElementUpdatable(ElementAttributes element, String userId, boolean forDeletion) {
-        if (element.getType().equals(DIRECTORY)) {
+        if (element.getType().equals(ElementType.DIRECTORY)) {
             return element.isAllowed(userId) &&
                 (!forDeletion || getDirectoryElementsStream(element.getElementUuid(), userId, List.of())
-                    .filter(e -> e.getType().equals(DIRECTORY))
+                    .filter(e -> e.getType().equals(ElementType.DIRECTORY))
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:388]: Use ==

**Pull request closed without merging**